### ISSUE TITLE
💅 noParameterAssign doesnt catch single parameter case without parens

### ISSUE THREAD

**Issue by pbarbiero** [biomejs/biome#5409]
### Environment information

```bash
CLI:
  Version:                      1.9.4
  Color support:                true

Linter:
  JavaScript enabled:           true
  Enabled rules:
  lint/correctness/noParameterAssign
```

### Rule name

noParameterAssign

### Playground link

https://biomejs.dev/playground/?arrowParentheses=as-needed&lintRules=all&code=YwBvAG4AcwB0ACAAZABvAFMAbwBtAGUAdABoAGkAbgBnACAAPQAgAHIAZQBxACAAPQA%2BACAAewAKACAAIAByAGUAcQAgAD0AIAB7AH0AOwAgAC8ALwAgAGQAbwBlAHMAbgB0ACAAZgBsAGEAZwAKAH0AOwAKAAoAYwBvAG4AcwB0ACAAZABvAFMAbwBtAGUAdABoAGkAbgBnADIAIAA9ACAAKAByAGUAcQAsACAAcgBlAHEAMgApACAAPQA%2BACAAewAKACAAIAByAGUAcQAgAD0AIAB7AH0AOwAgAC8ALwAgAGcAZQB0AHMAIABmAGwAYQBnAGcAZQBkAAoAfQA7AA%3D%3D

### Expected result

Rule triggers when reassigning even to a single parameter when Arrow Parenthesis is set to "As Needed"

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
@pbarbiero can you update the reproduction and enable all rules?
